### PR TITLE
422 add application status card to project details

### LINKED ISSUE #422: project details: add MVP application status
**User Story** 
As a project owner, 
I want to confirm whether or not I've applied to a round 
so that I know my application has been received 

**Acceptance Criteria**
GIVEN that I just applied to a round 
WHEN I am redirected to the Project Details page  
THEN I see the "My Applications" section with my application status 

The "My Applications" section should include: 
* Heading that reads "My Applications" 
* Program Name
* Round Name
* Round start and end dates (not application dates) 
* How to reach out to Grant Round support team 

**Product & Design Links:** 
* [Figma link](https://www.figma.com/file/obBzCCL8KwoTlVBb5g3ccN/Grant-Hub---Wireframes?node-id=2557%3A17537)
Screen Shot 2022-10-02 at 11 57 48 AM

**Tech Details**: 
* N/A 

**Open Questions**: 
* N/A

**Notes/Assumptions**: 
* The text and link to contact customer support will need to come from round metadata (Round Manager Dependency)

### PR DESCRIPTION
**Pull request by codenamejason**
Tasks Included:

- Adds "Application Status" Card in "Project Details" page

Todo: 

- [x] status tests 
- [x] Graph data to update Card values (depending on PR #469)
- [x] Use the application data to update `ApplicationCard.tsx`

*** @michellema1208  should we wait for Issue [#591](https://github.com/gitcoinco/grants-round/issues/591) from Round Manager for customer support contact info.

### CODE REVIEW

File: client/src/reducers/projects.ts [hunk 1]
```diff
@@ -10,48 +10,51 @@ import {
...
   status: Status.Undefined,
   error: undefined,
   ids: [],
   events: {},
-  applications: {
-    error: undefined,
-  },
+  applications: [
+    {
+      projectId: "",
```
Comment (DanieleSalatti) [inline client/src/reducers/projects.ts]: Do we need the `projectId` here, or is it going to be always the same one repeated? Looking at the actions it seems to be the latter?

File: client/src/actions/projects.ts [hunk 2]
```diff
@@ -136,30 +135,42 @@ export const loadProjects =
...
+        { projectID }
+      );
+      console.log("applicationsFound", applicationsFound);
+
+      dispatch({
+        type: PROJECT_APPLICATIONS_LOADED,
+        projectID,
+        applications: applicationsFound.data.roundProjects,
+      });
+    } catch (error: any) {
```
Comment (DanieleSalatti) [inline client/src/actions/projects.ts:163]: We should add datadog metrics/logging on exceptions

File: client/src/actions/projects.ts [hunk 3]
```diff
@@ -28,29 +28,28 @@ export interface ProjectsUnloadedAction {
...
   roundID: string;
 }
 
 export const PROJECT_APPLICATIONS_LOADED = "PROJECT_APPLICATIONS_LOADED";
 interface ProjectApplicationsLoadedAction {
   type: typeof PROJECT_APPLICATIONS_LOADED;
-  projectID: number;
-  applications: ApplicationStatus;
+  projectID: string;
+  applications: any;
```
Comment (DanieleSalatti) [inline client/src/actions/projects.ts:44]: Nit: we could make this typed

File: client/src/components/grants/ApplicationCard.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+import { useDispatch, useSelector } from "react-redux";
+import { loadRound, unloadRounds } from "../../actions/rounds";
+import { RootState } from "../../reducers";
+import { formatDate } from "../../utils/components";
+
+// type ApplicationCardProps = {
+//   projectID: string;
+//   chainId: number;
+//   applications: any;
+// };
```
Comment (DanieleSalatti) [inline client/src/components/grants/ApplicationCard.tsx]: Remove?

File: client/src/components/grants/ApplicationCard.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+          </Badge>
+        </Box>
+      </SimpleGrid>
+      <Box className="pl-2 text-gitcoin-gray-400">
+        <span>{renderRoundDate()}</span>
+      </Box>
+      <Box p={2} className="mt-4 mb-6">
+        <p>
+          Have any questions about your grant round application? Contact{" "}
+          <a className="text-purple-500" href="/">
```
Comment (DanieleSalatti) [inline client/src/components/grants/ApplicationCard.tsx:76]: QQ: are we keeping track of things like this that we will need to fill in the near future?
Might be worth creating a task/issue and adding a TODO comment here referencing it?

File: client/src/components/grants/Details.tsx [hunk 6]
```diff
@@ -59,88 +100,110 @@ export default function Details({
...
+              </a>
+            </div>
+          )}
+        </Box>
+        <Box>
+          <Box p={1}>
+            <span className="text-[20px]">My Applications</span>
+          </Box>
+          <Box>
+            {props.applications.length > 0 &&
```
Comment (DanieleSalatti) [inline client/src/components/grants/Details.tsx]: Won't this always be > 0? The initial state contains an empty one and I think that's what is causing to show an unknown application?

Schermata 2022-10-26 alle 08 48 44

**Review by DanieleSalatti — COMMENTED** [review]
Left some comments

File: client/src/components/grants/Details.tsx (hunk 6 above)
Comment (codenamejason) [reply client/src/components/grants/Details.tsx]: I updated the initial state value to []

File: client/src/components/grants/ApplicationCard.tsx (hunk 5 above)
Comment (codenamejason) [reply client/src/components/grants/ApplicationCard.tsx:76]: @michellema1208 can we create a ticket for this so we don't lose track?

Add link for product support coming from round manager data if it is added on their end as a required field or not.

**Review by michellema1208 — APPROVED** [review]

**Pull request merged**